Created electron guide (#6240)
expo/expo: docs/pages/versions/unversioned/guides/using-electron.md

---
title: Using Electron with Expo for Web
---

[![supports Windows](https://img.shields.io/badge/Windows-4630EB.svg?style=for-the-badge&logo=WINDOWS&labelColor=000&logoColor=fff)](https://github.com/expo/expo)
[![supports Linux](https://img.shields.io/badge/Linux-4630EB.svg?style=for-the-badge&logo=LINUX&labelColor=000&logoColor=fff)](https://github.com/expo/expo)
[![supports OSX](https://img.shields.io/badge/OSX-4630EB.svg?style=for-the-badge&logo=APPLE&labelColor=000&logoColor=fff)](https://github.com/expo/expo)

> > Notice: Electron support is experimental, so the workflow is suboptimal and subject to breaking changes. If you find bugs please report them on [expo/expo-cli](https://github.com/expo/expo-cli/issues) with the `[electron]` tag in the title.

Reviewer: ```suggestion
> 🚨 Electron support is experimental, so the workflow is suboptimal and subject to breaking changes. If you find bugs please report them on [expo/expo-cli](https://github.com/expo/expo-cli/issues) with the `[electron]` tag in the title.
```

> > Notice: Electron is an experimental feature with Expo so the workflow is suboptimal and subject to breaking changes. If you find bugs please report them on [expo/expo-cli](https://github.com/expo/expo-cli/issues) with the `[electron]` tag in the title.

Reviewer: ```suggestion
> Notice: Electron support is experimental, so the workflow is suboptimal and subject to breaking changes. If you find bugs please report them on [expo/expo-cli](https://github.com/expo/expo-cli/issues) with the `[electron]` tag in the title.
```

> [Electron][electron] is a framework for creating desktop apps that run in a Chromium wrapper. Using Expo with Electron will enable you to use your existing components to build OSX, Windows, and Linux apps!

Reviewer: ```suggestion
[Electron][electron] is a framework for creating desktop apps that run in a Chromium wrapper. Using Expo with Electron will enable you to use your existing components to build OSX, Windows, and Linux apps.
```

  module.exports = withExpoAdapter({
    projectRoot: __dirname,
    // Provide any overrides for electron-webpack: https://github.com/electron-userland/electron-webpack/blob/master/docs/en/configuration.md
  });
  ```

## ⚽️ Usage

### Starting a project

- Start the project with `yarn expo-electron start`
  - Currently this is an alias for the following script: `ELECTRON_DISABLE_SECURITY_WARNINGS=1 electron-webpack dev`
  - `ELECTRON_DISABLE_SECURITY_WARNINGS=1` disables the unused security warnings in your project.

### Customizing the main process

- To reveal the main process (highly recommended) run - `yarn expo-electron customize`
  - This will generate the `electron/main/` and `electron/webpack.config.js` files for you to customize.
  - Everything running in the `electron/main/` folder is on a different process to the rest of your app. Think of this like the native code in the Expo Client app (but not really because it's JavaScript and simple).
- To revert back to the default main process or reset to the latest default template simply delete the `electron/` folder and the adapter will go back to using the hidden version.

### Building your project

> `@expo/electron-adapter` doesn't do anything to streamline building Expo Electron projects just yet. But until it does here is a guide for building projects using [`electron-builder`][electron-builder].

Reviewer: ```suggestion
`@expo/electron-adapter` doesn't do anything to streamline the build phase of Electron projects just yet. But until it does here is a guide for building projects using [`electron-builder`][electron-builder].
```

- Use the builder with: `yarn electron-webpack && yarn electron-builder --dir -c.compression=store -c.mac.identity=null` (`-c.compression=store` speeds the builds up a lot, delete this for actual production builds)
- Learn more about configuring your build here: [Configuring electron-builder][electron-builder-config]

## 🧸 Behavior

- Webpack now resolves files with `.electron.js` & `.web.js` extensions in that order. If you want to use `electron` features then put them in a file like `foo.electron.js`.
- Every universal package you have installed will be transpiled automatically, this includes packages that start with the name: `expo`, `@expo`, `@unimodules`, `@react-navigation`, `react-navigation`, `react-native`. You can add more by appending them to the array for key `expo.web.build.babel.include` in your `app.json` (this feature is experimental and subject to change).

## Contributing

If you would like to help make Electron support in Expo better, please feel free to open a PR or submit an issue:

- [Expo CLI][expo-cli]

If you want to add first-class electron support to any of the Unimodules then you can submit PRs to the expo/expo repo:

> - [Expo packages][expo-packages]

Reviewer: ```suggestion
- [Expo SDK packages][expo-packages]
```
